### PR TITLE
8301628: RISC-V: c2 fix pipeline class for several instructions

### PR DESCRIPTION
**Pull request by zifeihan**
HI,

The current C2 instructions can use some more accurate ins_pipe, eg.:
```
instruct regI_not_reg(iRegINoSp dst, iRegI src1, immI_M1 m1) %{
  match(Set dst (XorI src1 m1));
  ins_cost(ALU_COST);
  format %{ "xori  $dst, $src1, -1\t#@regI_not_reg" %}

  ins_encode %{
    __ xori(as_Register($dst$$reg), as_Register($src1$$reg), -1);
  %}

  ins_pipe(ialu_reg);
%}
```

We can use the more accurate pipe_class `ialu_reg_imm` instead.

Please take a look and have some reviews. Thanks a lot.

## Testing:
- hotspot and jdk tier1 on unmatched board without new failures

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301628](https://bugs.openjdk.org/browse/JDK-8301628): RISC-V: c2 fix pipeline class for several instructions

### Reviewers
 * [Feilong Jiang](https://openjdk.org/census#fjiang) (@feilongjiang - Author) ⚠️ Review applies to [64666538](https://git.openjdk.org/jdk/pull/12379/files/64666538644bc93c5ba01d827cef39a6d89a6538)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12379/head:pull/12379` \
`$ git checkout pull/12379`

Update a local copy of the PR: \
`$ git checkout pull/12379` \
`$ git pull https://git.openjdk.org/jdk pull/12379/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12379`

View PR using the GUI difftool: \
`$ git pr show -t 12379`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12379.diff">https://git.openjdk.org/jdk/pull/12379.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/riscv/riscv.ad [hunk 1]
```diff
@@ -7308,59 +7308,67 @@ instruct minD_reg_reg(fRegD dst, fRegD src1, fRegD src2, rFlagsReg cr) %{
                  true /* is_double */, true /* is_min */);
   %}
 
-  ins_pipe(fp_dop_reg_reg_d);
+  ins_pipe(pipe_class_default);
 %}
 
 // Float.isInfinite
 instruct isIniniteF_reg_reg(iRegINoSp dst, fRegF src)
```
Comment (feilongjiang) [inline src/hotspot/cpu/riscv/riscv.ad:7315]: Here is a typo and could you please fix it too?
s/isIniniteF_reg_reg/isInfiniteF_reg_reg

**Review by feilongjiang — APPROVED** [review]
Looks good.

File: src/hotspot/cpu/riscv/riscv.ad (hunk 1 above)
Comment (zifeihan) [reply src/hotspot/cpu/riscv/riscv.ad]: > Here is a typo and could you please fix it too? s/isIniniteF_reg_reg/isInfiniteF_reg_reg

Thanks for your review, done.

**Review by RealFYang — APPROVED** [review]

**Comment by zifeihan** [comment]
Thanks all for the review.
/integrate

**Pull request closed without merging**